index.html

```diff
@@ -2595,6 +2595,119 @@ <h3>Data Schemas</h3>
+          resource.
+        </p>
+        <p>
+          Each object in the
+          <code>resourceIntegrity</code> array MUST contain the following:
+          the URL to the resource named <code>resource</code>, a
+          <code>timestamp</code>
+          that indicates the time at which the hash was computed, the
+          <code>hash</code>
+          of the resource, and the <code>method</code> which indicates what
+          hashing algorithm was used as listed as the 'Hash Name String'
+          property from the <a
+          href="https://www.iana.org/assignments/named-information/named-information.xhtml">IANA
+          Named Information Hash Algorithm Registry</a>.  
+          The <code>hash</code> property MUST be the base64url [[RFC 4648]]
+          encoded digest of the hash.
+          The <code>timestamp</code> property MUST be a string value of an
+          [[XMLSCHEMA11-2]] combined date-time string. An implementer may
+          include other fields in each object.
+        </p>
+        <p>
+          An object in the <code>resourceIntegrity</code> array MAY contain
+          a property named <code>mediaType</code> that indicates the 
+          expected media type for the indicated <code>resource</code>.
+          If a <code>mediaType</code> is included it must be a valid
+          media type as listed in the 
+          <a href="https://www.iana.org/assignments/media-types/media-types.xhtml">
+            IANA Media Types
+          </a> registry.
+        </p>
+        <p>
+          An object in the <code>resourceIntegrity</code> array MAY contain
+          additional properties.
+        </p>
+        <p>
+          Implementers should consult appropriate sources, such as the <a
+          href="https://www.iana.org/assignments/named-information/named-information.xhtml">IANA
+          Named Information Hash Algorithm Registry</a> to ensure that they
+          are chosing a current and reliable hash algorithm.  At the time of
+          this writing `sha-256` should be considered the minimum strength
+          hash algorithm for use by implemnters.
+        </p>
+        <p class="note">
+          If at a later date subresource integrity as defined in [[SRI]] is
+          adopted into the [[JSON-LD]] specification as noted in that
+          specifications <a
+          href="https://www.w3.org/TR/json-ld11/#security">current security
+          considerations</a> of that specification, this hash in the VC can
+          serve as an additional check towards ensuring that a cached
+          context used when issuing the VC matches the remote resource.
+        </p>
+        <p>
+          <aside
+            class="example"
+            title="resource integrity"
+          >
+            <p>An example of a resource integrity object referencing contexts</p>
+            <pre>
+              "resourceIntegrity": [{
+                "resource": "https://www.w3.org/ns/credentials/v2",
```

Comment:
Note: If we use `id` here instead of `resource` and allow `digestMultibase` as an option to express the hash information, then we'd have something almost identical to what was proposed in #831. It would be good to understand pros/cons for using `resource` vs. `id`.

The advantages of enabling multibase-encoded digest values that use multiformats (like canon-hash) has already been mentioned; a disadvantage would be if we have two different ways of expressing hash information.

Breaking out the hash parameters in JSON (as opposed to encapsulating in a multiformats style expression) requires application-level JSON schemas to be more complicated and spread the validation burden around vs. confining it to the integrity checking library.

These are things we thought of in the past when a number of implementations were created that used the approach in #831, including some solutions that use it in production systems now. I'm concerned we're not reusing those lessons or even addressing them here with this new approach that, from all indications, was reactively generated to be responsive to (perfectly reasonable) security considerations.

Reply:
I'm not sure we should mandate multiformats, or if we are even allowed to... seems like a downref.

I suggest, we use IANA for now.

We could make this more concrete by defining a `type` that does not use multiformats, and then a new type could be used in multiformats become standard in the future.